hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/HermesServer.java

```diff
@@ -101,10 +135,30 @@ private HttpHandler handlers() {
                 .get("/status/health", healthCheckHandler)
                 .get("/", healthCheckHandler);
 
-        return isEnabled(FRONTEND_REQUEST_DUMPER) ? new RequestDumpingHandler(routingHandler) : routingHandler;
+        return isEnabled(FRONTEND_REQUEST_DUMPER) ? new RequestDumpingHandler(routingHandler) :
+                isEnabled(FRONTEND_AUTHENTICATION_ENABLED) ? withAuthenticationHandlersChain(routingHandler) : routingHandler;
+    }
+
+    private HttpHandler withAuthenticationHandlersChain(HttpHandler next) {
```

Comment:
We can surround body of this function with try-catch to handle all potentially unreadable exceptions like valueOf for AuthenticationMode.